src/deepsparse/middlewares/middleware.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Description:
```

Comment:
Very nice description, I have a small suggestion, could we structure this like the following:
```
Description:
     Middlewares are used ...
Lifecycle:
    Pipeline -> Middleware ...
...
```

I think the indentation would make it look much nicer? Thoughts?